app/__init__.py

```diff
@@ -147,7 +149,31 @@ def route(self, rule, **options):
         and POST methods, useful for form-based routes.
         """
         options.setdefault("methods", ("GET", "POST"))
-        return super().route(rule, **options)
+        
+        def decorator(func):
+            # Get function signature
+            signature = inspect.signature(func)
+            annotations = signature.parameters
+            
+            def wrapped_view(*args, **kwargs):
+                # Inject forms into kwargs based on annotations
+                for name, param in annotations.items():
+                    if issubclass(param.annotation, FlaskForm):
```

Comment:
When all of the logic of your loop is guarded in an if-condition, try to revert your if instead:

```py
for x in something:
  if x == 1:
    ...
```
becomes
```py
for x in something:
  if x != 1:
    break
  ...
```

by removing one level of indentation, you reduce complexity and makes it easier to follow, the reader does not have to remember that there is a condition going on, and does not have to check if there is an "else" or other "ifs"

so here it would be:
```py
if not issubclass(param.annotation, FlaskForm):
  continue
form_class = param.annotation
...
```